HW0/P5/P5.txt

```diff
@@ -1 +1,41 @@
+
+6. Also assume it takes one second to hand a bag (or any number of bags) between two people (including from you to a cashier). How long will it take to verify 256 bags worth of coins if each cashier comes to you and takes 1 bag? What about 2 bags? Is there a better strategy? Assume you can’t be handing bags to more than one person at a time.
+
+In order to verify the 256 bags worth coins, given that it takes exactly 1 second to hand a bag of coins to a cashier we need to examine how long it takes to hand all of the bags to the employees. In the case of 8 employees working, it will take exactly 8 seconds to hand over the first round of bags, then an additional 8 seconds to hand of the next round and so on and so forth. The trick here is that one employee can start summing their bags as you hand of bags to the next employee. So to hand over 2 rounds of bags, and to allow time for all employees to add their bags, it will take 8+8+1=17 seconds. If we apply this logic to all bags, it takes 17 seconds for 2 rounds or 16 bags, it will take 16 rounds of 16 bags to complete the task or 16*16+1 = 257. Now for the communication costs (log_2(E)) and the additional time needed to add (log_2(E)) we add the 2 * log_2(E) factor and arrive at 257 + 2 * log_2(8) = 256+(2*3) = 263.
+
+Now we can assume that it takes the same amount of time to hand of 2 bags as it does to hand off 1 bag. This means that employees can start summing their bags as I start to hand of bags to the next cashier which should result in reduced verification time. If we once again have 8 employees and 256 bags, I will hand 2 bags to the first employee and while I hand over the next 2 bags, the first employee will start summing. Now it will take B/2 time for each employee to count his or her bag. So for 256/2 = 128 seconds for each employee to count and since the employees are counting in a cascading manner, we must add 1 additional second to account for the time that the last employee needs to complete counting. Adding in our communication and employee constant of 2*log_2(E) arriving at the final answer of (256/2) + 1 + 2 * log2(8) = 128 + 2 + 2*3 = 135 seconds.
+
```

Comment:
Again using the same idea I mentioned above, I think the previous employees can count and communicate while the next employees are receiving bags so I think after the last employee receives his bags it takes 7 seconds (if he can count by 2s) for him to count, 1 second to communicate, and 1 second to add. So it could be 128+9 = 137.